include/swift/Basic/SwiftRuntime.h

```diff
@@ -0,0 +1,54 @@
+//===--- SwiftRuntime.h - Swift Runtime Handling ----------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_SWIFTRUNTIME_H
+#define SWIFT_BASIC_SWIFTRUNTIME_H
+
+#include "llvm/ADT/Triple.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Support/VersionTuple.h"
+
+namespace swift {
+
+class SwiftRuntime {
+  VersionTuple Version;
+
+public:
+  SwiftRuntime(const VersionTuple &V) : Version(V) {}
+
+  const VersionTuple &getVersion() const { return Version; }
+
+
+  bool hasDefaultForeignClassValueWitnesses(const llvm::Triple &T) const {
+    return Version >= VersionTuple(5, 1) || !T.isOSDarwin();
+  }
```

Comment:
Just having a target runtime version is sufficient.  The way this should work is that we should construct this with either:
- the value of the `-swift-runtime-version` flag, if provided; or if not that,
- the version of the system Swift runtime for the target OS, if the target has a system Swift runtime (i.e. if it is Darwin); or if not that,
- the current compiler version, because we expect to link with the bundled Swift runtime.